mlflow/pyfunc/model.py

```diff
@@ -163,12 +195,22 @@ class _FunctionPythonModel(PythonModel):
     in an instance of this class.
     """
 
+    _skip_wrapping_predict = True
+
     def __init__(self, func, signature=None):
-        self.func = func
         self.signature = signature
+        # only wrap `func` if @pyfunc is not already applied
+        if not getattr(func, "_is_pyfunc", False):
+            self.func = pyfunc(func)
+        else:
+            self.func = func
 
-    def _get_type_hints(self):
-        return _extract_type_hints(self.func, input_arg_index=0)
+    @property
```

Comment:
nit: do we need this override?

Reply:
Yes because predict function doesn't have type hints, we're extracting type hints from self.func

Reply:
Ah makes sense!